Source/ASTextNode.mm

```diff
@@ -1133,36 +1160,47 @@ - (UIEdgeInsets)shadowPaddingWithRenderer:(ASTextKitRenderer *)renderer
 
 - (void)setTruncationAttributedText:(NSAttributedString *)truncationAttributedText
 {
-  ASDN::MutexLocker l(__instanceLock__);
-  
-  if (ASObjectIsEqual(_truncationAttributedText, truncationAttributedText)) {
-    return;
+  {
+    ASDN::MutexLocker l(__instanceLock__);
+    
+    if (ASObjectIsEqual(_truncationAttributedText, truncationAttributedText)) {
+      return;
+    }
+
+    _truncationAttributedText = [truncationAttributedText copy];
   }
 
-  _truncationAttributedText = [truncationAttributedText copy];
   [self _invalidateTruncationText];
```

Comment:
I audit all of the methods and I changed it to `_locked_` and remove holding the lock.